doc/creativision.sgml

```diff
@@ -0,0 +1,174 @@
+<!doctype linuxdoc system>
+
+<article>
+
+<title>VTech Creativision (aka Funvision) specific information for cc65
+<author><url url="mailto:polluks+cc65@sdf.lonestar.org" name="Stefan A. Haubenthal">
+<date>2016-04-14
+
+<abstract>
+An overview over the Creativision runtime system as it is implemented for the
+cc65 C compiler.
+</abstract>
+
+<!-- Table of contents -->
+<toc>
+
+<!-- Begin the document -->
+
+<sect>Overview<p>
+
+This file contains an overview of the Creativision runtime system as it comes
+with the cc65 C compiler. It describes the memory layout, Creativision specific header
+files, available drivers, and any pitfalls specific to that platform.
+
+Please note that Creativision specific functions are just mentioned here, they are
+described in detail in the separate <url url="funcref.html" name="function
+reference">. Even functions marked as "platform dependent" may be available on
+more than one platform. Please see the function reference for more information.
+
+
+<sect>Binary format<p>
+
+The standard binary output format generated by the linker for the Creativision target
+is a 4 kbyte machine language program. It is of course possible to change
+this behaviour by using one of the different linker configs.
+
+<sect>Memory layout<p>
+
+cc65 generated programs with the default setup run with the I/O area enabled,
+which gives a usable memory range of &dollar;B000 - &dollar;BEFF.
+More ROM may need additional bankswitching code.
+
+Special locations:
+
+<descrip>
+  <tag/Text screen/
+  The text screen is located at VRAM &dollar;1000.
+
+  <tag/Stack/
+  The C runtime stack is located at &dollar;3FF and growing downwards.
+
+  <tag/Heap/
+  The C heap is located at the end of the program and grows towards the C
+  runtime stack.
+
+</descrip><p>
+
+
+
+<sect>Platform specific header files<p>
+
+Programs containing Creativision specific code may use the <tt/creativision.h/ header file.
+
+
+<sect1>Creativision specific functions<p>
+
+<itemize>
+<item>bios_playsound
+<item>joystate
+<item>psg_delay
+<item>psg_outb
+<item>psg_silence
+</itemize>
+
+
+
+<!--<sect1>Hardware access<p>
+
+The following pseudo variables declared in the <tt/creativision.inc/ include file do
+allow access to hardware located in the address space.
+
+<descrip>
+
+  <tag><tt/VDP/</tag>
+  The <tt/VDP/ defines allow access to the video chip.
+
+</descrip><p>
+
+<descrip>
+
+  <tag><tt/PIA/</tag>
+  The <tt/PIA/ defines allow access to the I/O chip.
+
+</descrip><p>-->
+
+
+
+<sect>Loadable drivers<p>
+
+<sect1>Graphics drivers<p>
+
+No graphics drivers are currently available for the Creativision.
+
+
+<sect1>Extended memory drivers<p>
+
+No extended memory drivers are currently available for the Creativision.
+
+
+<sect1>Joystick drivers<p>
+
+No joystick drivers are currently available for the Creativision but <tt/joystate()/.
```

Comment:
I get notifications -- for example:
> Subject: Re: [cc65/cc65] creativision target (#374)
> From: Christian Groessler <notifications@github.com>
> Date: 2017-02-06 3:06 PM
> To: cc65/cc65 <cc65@noreply.github.com>
>
> @groessler pushed 2 commits.
>    - e6359ec Add initial creativision.sgml, contributed by Stefan Haubenthal.
>    - c33383e Add Creativision to README.md and html index
>
> —
> You are receiving this because you are subscribed to this thread.
> View it on GitHub or mute the thread.